### ISSUE TITLE
ECR client no longer respects the endpoint_url parameter

### ISSUE THREAD

**Issue by bblommers** [boto/botocore#3326]
### Describe the bug

When invoking ECR methods on a client with the `endpoint_url` configured, the configured endpoint is ignored, and all requests are actually send to `ecr.amazonaws.com`.

### Regression Issue

- [X] Select this option if this issue appears to be a regression.

### Expected Behavior

 I would expect the request to go to that endpoint url. This worked with botocore `1.35.87`, so this is a recent regression.

### Current Behavior

```
An error occurred (UnrecognizedClientException) when calling the CreateRepository operation: The security token included in the request is invalid.
```

This error tells me that the requests are actually send through to AWS, and not to the configured endpoint. (The endpoint in my case is Moto, which accepts any request, regardless of the security token included.)

### Reproduction Steps

```
import os

import boto3
from botocore.exceptions import EndpointConnectionError

os.environ["AWS_ACCESS_KEY_ID"] = 'testing'
os.environ["AWS_SECRET_ACCESS_KEY"] = 'testing'
os.environ["AWS_DEFAULT_REGION"] = 'us-east-1'


def test_create_repository():
    # given
    client = boto3.client("ecr", endpoint_url="http://localhost:5000")

    try:
        client.create_repository(repositoryName="myrepo")
    except EndpointConnectionError as e:
        # Expected, as nothing is running on localhost
        pass
```

Running this test passes against `1.35.87`, as expected, but fails against `1.35.88` with an `UnrecognizedClientException` - because botocore is trying to reach out to AWS instead.

### Additional Information/Context

I've created a very simple GitHub repo if you want to see the above test in action, against both `botocore` versions:
https://github.com/bblommers/botocore-ecr-endpoint/pull/1

Note that also fails against other methods, like DescribeRepositories and PutReplicationConfiguration

### SDK version used

1.35.88

### Environment details (OS name and version, etc.)

Ubuntu

**Comment by SamRemis** [comment]
@bblommers,

Thank you for bringing this to our attention.  We have confirmed the issue and are working on a fix

**Comment by bblommers** [comment]
Looks like this is already fixed with `1.35.90`. Thank you for looking into this and for the quick fix @SamRemis!
